src/router.ts

```diff
@@ -17,27 +18,41 @@ export function createRouter<T extends RadixNodeData = RadixNodeData>(
     staticRoutesMap: {},
   };
 
-  const normalizeTrailingSlash = (p) =>
+  const normalizeTrailingSlash = (p: string) =>
     options.strictTrailingSlash ? p : p.replace(/\/$/, "") || "/";
 
   if (options.routes) {
     for (const path in options.routes) {
-      insert(ctx, normalizeTrailingSlash(path), options.routes[path]);
+      Array.isArray(options.routes[path]) &&
+      options.routes[path].length === 2 &&
+      HTTPMethods.includes(options.routes[path][0])
+        ? insert(
+            ctx,
+            normalizeTrailingSlash(path),
+            options.routes[path][1],
+            options.routes[path][0]
+          )
+        : insert(ctx, normalizeTrailingSlash(path), options.routes[path]);
     }
   }
 
   return {
     ctx,
-    // @ts-ignore
-    lookup: (path: string) => lookup(ctx, normalizeTrailingSlash(path)),
-    insert: (path: string, data: any) =>
-      insert(ctx, normalizeTrailingSlash(path), data),
+    // @ts-expect-error - types are not matching
+    lookup: (path: string, options?: { method?: HTTPMethod }) =>
+      lookup(ctx, normalizeTrailingSlash(path), options?.method),
```

Comment:
We could directly pass entire options object forward to avoid future refactors. Also maybe a type like `LookupOptions` to share.